server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexAliasesService.java

```diff
@@ -133,7 +133,7 @@ ClusterState innerExecute(ClusterState currentState, Iterable<AliasAction> actio
                     Function<String, IndexMetaData> indexLookup = name -> metadata.get(name);
                     aliasValidator.validateAlias(alias, action.getIndex(), indexRouting, indexLookup);
                     if (Strings.hasLength(filter)) {
-                        IndexService indexService = indices.get(index.getIndex());
```

Comment:
There's definitely something wrong here: `indices` is a `Map<String, IndexService>` so `indices.get()` takes a `String`, but `index.getIndex()` returns an `Index`. However, I'm not sure why this fix is the right thing to do. This needs another reviewer.

Reply:
In my original review I stared at this for a long time. And then I looked at it in IntelliJ and realized there is a warning. Java isn't strict about the types here. I think what this means is we always get null back right now, and so always put something back into the indices map.

Reply:
Yes, that agrees with my analysis from the commit message:

> a get of the wrong type would
> always have returned null so these indices would have been
> inserted into the map repeatedly.